GCC 8 or 7 with Mac?

On the [installation instructions](https://xgboost.readthedocs.io/en/latest/build.html), conflicting accounts are given for multithreading on Mac with R:

> First, obtain gcc-8 with Homebrew (https://brew.sh/) to enable multi-threading (i.e. using multiple CPU threads for training). The default Apple Clang compiler does not support OpenMP, so using the default compiler would have disabled multi-threading.
> 
> brew install gcc@8
> Now, clone the repository:
> 
> git clone --recursive https://github.com/dmlc/xgboost
> Create the build/ directory and invoke CMake with option R_LIB=ON. Make sure to add CC=gcc-8 CXX=g++-8 so that Homebrew GCC is selected. After invoking CMake, you can install the R package by running make and make install:
> 
> mkdir build
> cd build
> CC=gcc-7 CXX=g++-7 cmake .. -DR_LIB=ON
> make -j4
> make install

The page recommends installing gcc8 via brew, but then proceeds to use gcc7 in the installation instructions. I've tried both, and each throws an error at 90% that says:

> [ 90%] Building CXX object src/CMakeFiles/objxgboost.dir/tree/updater_sync.cc.o
> [ 90%] Built target objxgboost
> make: *** [all] Error 2

update - been working on this for a couple hours, and 0 progress. Not convinced it is easy to have multi-threading support on Mac with R at this time.

--- a/doc/build.rst
+++ b/doc/build.rst
@@ -375,7 +375,7 @@ Create the ``build/`` directory and invoke CMake with option ``R_LIB=ON``. Make 
 
   mkdir build
   cd build
-  CC=gcc-7 CXX=g++-7 cmake .. -DR_LIB=ON
+  CC=gcc-8 CXX=g++-8 cmake .. -DR_LIB=ON
   make -j4
   make install
 

--- a/python-package/.pylintrc
+++ b/python-package/.pylintrc
@@ -4,7 +4,7 @@ ignore=tests
 
 extension-pkg-whitelist=numpy
 
-disable=unexpected-special-method-signature,too-many-nested-blocks,useless-object-inheritance
+disable=unexpected-special-method-signature,too-many-nested-blocks,useless-object-inheritance,import-outside-toplevel,unsubscriptable-object
 
 dummy-variables-rgx=(unused|)_.*
 

--- a/python-package/xgboost/core.py
+++ b/python-package/xgboost/core.py
@@ -984,7 +984,7 @@ class DMatrix(object):
             # validate feature name
             try:
                 if not isinstance(feature_names, str):
-                    feature_names = [n for n in iter(feature_names)]
+                    feature_names = list(feature_names)
                 else:
                     feature_names = [feature_names]
             except TypeError:
@@ -1028,7 +1028,7 @@ class DMatrix(object):
 
             try:
                 if not isinstance(feature_types, str):
-                    feature_types = [n for n in iter(feature_types)]
+                    feature_types = list(feature_types)
                 else:
                     feature_types = [feature_types]
             except TypeError:

--- a/python-package/xgboost/tracker.py
+++ b/python-package/xgboost/tracker.py
@@ -153,8 +153,7 @@ class RabitTracker(object):
             except socket.error as e:
                 if e.errno in [98, 48]:
                     continue
-                else:
-                    raise
+                raise
         sock.listen(256)
         self.sock = sock
         self.hostIP = hostIP

